src/dvc_data/hashfile/state.py

```diff
@@ -98,10 +98,15 @@ def save(self, path, fs, hash_info):
         if not isinstance(fs, LocalFileSystem):
             return
 
+        if fs.version_aware:
+            _, version_id = fs.path.split_version(path)
```

Comment:
The version id should probably be extracted from info call (same as `size` and `etag` in the future).

Reply:
Looks like we might want to treat path with version_id as a complete path here, because you might have a situation where you have `s3://bucket/path` and `s3://bucket/path?version_id=12434` at the same time and state should actually care about both. Hm...

Reply:
> Also `version_aware` is only present for some filesystems, you'll get an attribute error.

This is addressed in the open dvc-objects PR https://github.com/iterative/dvc-objects/pull/133/files#